pkg/mcp/tool_filter.go

```diff
@@ -461,31 +494,86 @@ func processToolsListResponse(
 			return errToolNameNotFound
 		}
 
+		// Get description if present (optional in MCP spec)
+		description, _ := tool["description"].(string)
+
+		simpleTools = append(simpleTools, SimpleTool{
+			Name:        toolName,
+			Description: description,
+		})
+		toolMaps = append(toolMaps, tool)
+	}
+
+	// Apply the shared filtering/override logic
+	processedTools := applyFilteringAndOverrides(config, simpleTools)
+
+	// Build the filtered response by matching processed tools with their original maps
+	filteredTools := make([]map[string]any, 0, len(processedTools))
+	for _, processed := range processedTools {
+		// Find the original tool map by matching names
+		for i, simple := range simpleTools {
+			if simple.Name == processed.Name || simple.Name == findOriginalName(config, processed.Name) {
```

Comment:
this looks as high O complexity, but i do not think is critical, if we just have a few items. Will be a problem with a big list of tools